### PR TITLE
CORE-1016 Add documentation for `login_token`s - (Don't merge until after release 74)

### PR DESCRIPTION
**Pull request by tombusby**
**Don't merge until after release 74**

Questions:

- Many endpoints have a note explaining what permissions are required (example below). What is the correct way to state that a user needs `Update` privileges on another user to create a `login_token` for them?

> _OAuth Privileges required: `DOC_CREATE`_

- What is the ideomatic way to have a hyperlink to another endpoint in the docs. It would be helpful for `getpersonaltoken` and `gettokenforpersonalcredentials` to explicitly link to each other in their definitions. I can't find a good example of that so asking seems quicker than digging.

---
This change is [Reviewable](https://reviewable.io/reviews/scrive/scrive-apidocs/48)

### CODE REVIEW

**Comment by trin-cz** [comment]
@jonathanjouty volunteered to do the review.

**Comment by jonathanjouty** [comment]
As for the questions:

> * Many endpoints have a note explaining what permissions are required (example below). What is the correct way to state that a user needs `Update` privileges on another user to create a `login_token` for them?

AFAIK I don't think we have that defined anywhere currently...

> * What is the ideomatic way to have a hyperlink to another endpoint in the docs.

There isin't!
That's because we're generating Markdown, but Slate is generating HTML and thus the `#anchor` links. One way would be to see what it generates and then use that, but it's brittle.
I'd say let's just skip that for now :)

File: documentation/api/user/gettokenforpersonalcredentials.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  tags:
+    - User
+
+  description: |
+    Generate a temporary `login_token` which can be used with `getpersonaltoken`
+    to get valid OAuth credentials for that user. The endpoint also returns a
+    base64-encoded PNG image of a QR code. This QR code also contains the
+    `login_token` embedded in a simple JSON structure. The only purpose of the
+    JSON is to allow the potential for versioning in future.
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/user/gettokenforpersonalcredentials.yaml]: Done, we should add a note to that task's JIRA issue to update this when it's released.

File: documentation/api/user/gettokenforpersonalcredentials.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      required: false
+      default: 5
+      description: |
+        How many minutes the `login_token` should be valid for (maximum of 30).
+
+  responses:
+    200:
+      description: |
+        Returns a JSON containing `login_token`, `qr_code` and
+        `expiration_time`.
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/user/gettokenforpersonalcredentials.yaml]: No probs, will sort this in the morning.

[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/user/gettokenforpersonalcredentials.yaml]: Done (at least I think it's correct, you might want to double check).

File: documentation/api/user/gettokenforpersonalcredentials.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanjouty) [reply documentation/api/user/gettokenforpersonalcredentials.yaml]: I added a Slack reminder for you!

[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/user/gettokenforpersonalcredentials.yaml]: cheers :)

**Pull request merged**